docs/DeveloperGuide.md

```diff
@@ -250,26 +250,123 @@ Priorities: High (must have) - `* * *`, Medium (nice to have)  - `* *`,  Low (un
 Priority | As a ... | I want to ... | So that I can...
 -------- | :-------- | :--------- | :-----------
 `* * *` | new user | see usage instructions | refer to instructions when I forget how to use the App
-`* * *` | user | add a new person |
-`* * *` | user | delete a person | remove entries that I no longer need
-`* * *` | user | find a person by name | locate details of persons without having to go through the entire list
-`* *` | user | hide [private contact details](#private-contact-detail) by default | minimize chance of someone else seeing them by accident
-`*` | user with many persons in the address book | sort persons by name | locate a person easily
-
-{More to be added}
+`* * *` | user | add a new task using natural language | add a task to CMDo
+`* * *` | user | delete a task | remove entries that I no longer need
+`* * *` | user | edit a task| edit the task by accessing the task and typing the changes
+`* * *` | user | find a task by keywords | access details of tasks quickly without having to go through the entire list
+`* * *` | user |only view uncompleted task in CMDo| avoid being confused by completed task that are overdue| 
+`* * *` | user | my tasks sorted by due date and due time|locate urgent tasks easily | 
+`* * *` | user | remove my completed tasks| see only uncompleted tasks | 
+`* * *` | user | book by time slots| block out time for unconfirmed events |
+`* * *` | user | undo my previous action | make mistakes | 
+`* * *` | user | redo my previous action upon undo | make mistakes |
+`* * *` | user | set priority to my task | know which task is more important | 
+`* *` | user |Simply assign a date due to my todo by typing in the due date| to enter due dates easily
+`* *` | user |block out time slots of unconfirmed tasks| avoid scheduling tasks that clash
+`* *` | user |auto reschedule a task i am unable to complete due at the moment| save the effort of manual rescheduling
 
 ## Appendix B : Use Cases
 
-(For all use cases below, the **System** is the `AddressBook` and the **Actor** is the `user`, unless specified otherwise)
+(For all use cases below, the **System** is `CMDo` and the **Actor** is the `user`, unless specified otherwise)
+
+### Use case: Add a task
+
+**MSS**
+
+1. User requests to add a task
+2. CMDo stores the natural language into data
```

Comment:
Do note that in MSS ~ you need to have to have interaction back and forth (do users really need to know it stores the natural laguage into data?) do note that in Use cases, it defines the sequnce of interaction btw the role (user in your case) and the system for a succession event (goal) - Hence MSS :)